Adding allo

anmorrissey opened the pull request:
Describe your pull request here

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

### General instructions

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

### Instructions for avoiding API, ABI, and CLI breakage issues
Conda is able to record and lock (a.k.a. pin) dependency versions used at build time of other recipes.
This way, one can avoid that expectations of a downstream recipe with regards to API, ABI, or CLI are violated by later changes in the recipe.
If not already present in the meta.yaml, make sure to specify `run_exports` (see [here](https://bioconda.github.io/contributor/linting.html#missing-run-exports) for the rationale and comprehensive explanation).
Add a `run_exports` section like this:

```yaml
build:
  run_exports:
    - ...

```

with `...` being one of:

| Case                             | run_exports statement                                               |
| -------------------------------- | ------------------------------------------------------------------- |
| semantic versioning              | `{{ pin_subpackage("myrecipe", max_pin="x") }}`     |
| semantic versioning (0.x.x)      | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}`   |
| known breakage in minor versions | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| known breakage in patch versions | `{{ pin_subpackage("myrecipe", max_pin="x.x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| calendar versioning              | `{{ pin_subpackage("myrecipe", max_pin=None) }}`    |

while replacing `"myrecipe"` with either `name` if a `name|lower` variable is defined in your recipe or with the lowercase name of the package in quotes.

### Bot commands for PR management

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

Note that the <code>@BiocondaBot please merge</code> command is now depreciated. Please just squash and merge instead.

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

anmorrissey commented:
@BiocondaBot please add label

bgruening commented on recipes/allo/meta.yaml:
```diff
@@ -0,0 +1,44 @@
+    - pip
+    - python ==3.10
+  run:
+    - python ==3.10
+    - numpy ==1.23
+    - joblib ==1.1.0
+    - tensorflow ==2.11
+    - pysam ==0.20.0
```
```suggestion
    - python
    - numpy
    - joblib
    - tensorflow
    - pysam
```

Please do not use strict-pins. If pins are necessary, please use lower and upper bounds.

bgruening commented on recipes/allo/meta.yaml:
```diff
@@ -0,0 +1,44 @@
+  run_exports:
+    - {{ pin_subpackage('allo', max_pin="x") }}
+  number: 0
+  noarch: generic
```
```suggestion
  noarch: python
```

anmorrissey commented:
I haven't had a chance since yesterday to update the recipe after the first review. I was having issues with ranges when I was trying to add to conda-forge (irrelevant now) and forgot to change things back is all that happened. I will say tensorflow is going to end up strict through. It breaks extremely easily when I load in the models. I will test it this morning to find which versions of tensorflow will work and fix the recipe.

anmorrissey commented:
I updated the dependencies. Sorry for having them so stringent. I had to keep tensorflow stringent though because it easily breaks when you try to load in models from even other minor versions. I tested up until the newest version and it works fine so I changed the dependencies to match that.
